tests/e2e/progressive-e2e/progressive_test.go

```diff
@@ -0,0 +1,190 @@
+		updatedMonoVertexSpec.Source.UDTransformer.Container.Image = validUDTransformerImage
+
+		rawSpec, err = json.Marshal(updatedMonoVertexSpec)
+		Expect(err).ShouldNot(HaveOccurred())
+		UpdateMonoVertexRolloutInK8S(monoVertexRolloutName, func(mvr apiv1.MonoVertexRollout) (apiv1.MonoVertexRollout, error) {
+			mvr.Spec.MonoVertex.Spec.Raw = rawSpec
+			return mvr, nil
+		})
+
+		VerifyMonoVertexRolloutScaledDownForProgressive(monoVertexRolloutName, 0, monoVertexScaleTo, int64(monoVertexScaleMin), monoVertexScaleMinMaxJSONString, monoVertexScaleTo)
+		VerifyMonoVertexRolloutProgressiveStatus(monoVertexRolloutName, 0, 2, false, apiv1.AssessmentResultSuccess, defaultStrategy.Progressive.ForcePromote)
+
+		VerifyVerticesPodsRunning(Namespace, fmt.Sprintf("%s-%d", monoVertexRolloutName, 2),
+			[]numaflowv1.AbstractVertex{{Scale: updatedMonoVertexSpec.Scale}}, ComponentMonoVertex)
+		VerifyVerticesPodsRunning(Namespace, fmt.Sprintf("%s-%d", monoVertexRolloutName, 1),
+			[]numaflowv1.AbstractVertex{{Scale: numaflowv1.Scale{Min: ptr.To(int32(0)), Max: ptr.To(int32(0))}}}, ComponentMonoVertex)
+
+		DeleteMonoVertexRollout(monoVertexRolloutName)
+	})
+
+	It("Should validate MonoVertex upgrade using Progressive strategy via Forced Promotion - Failure case", func() {
+		By("Creating a MonoVertexRollout with ForcePromote enabled")
+		strategy := defaultStrategy.DeepCopy()
+		strategy.Progressive.ForcePromote = true
+		CreateMonoVertexRollout(monoVertexRolloutName, Namespace, initialMonoVertexSpec, strategy)
+
+		By("Verifying that the MonoVertex spec is as expected")
+		VerifyMonoVertexSpec(Namespace, monoVertexRolloutName, func(retrievedMonoVertexSpec numaflowv1.MonoVertexSpec) bool {
+			return reflect.DeepEqual(retrievedMonoVertexSpec, initialMonoVertexSpec)
+		})
+		VerifyInProgressStrategy(monoVertexRolloutName, apiv1.UpgradeStrategyNoOp)
+		VerifyMonoVertexRolloutHealthy(monoVertexRolloutName)
+
+		By("Updating the MonoVertex Topology to cause a Progressive change - Force promoted failure into success")
+		updatedMonoVertexSpec := initialMonoVertexSpec.DeepCopy()
+		updatedMonoVertexSpec.Source.UDTransformer = &udTransformer
+		updatedMonoVertexSpec.Source.UDTransformer.Container.Image = invalidUDTransformerImage
+
+		rawSpec, err := json.Marshal(updatedMonoVertexSpec)
+		Expect(err).ShouldNot(HaveOccurred())
+		UpdateMonoVertexRolloutInK8S(monoVertexRolloutName, func(mvr apiv1.MonoVertexRollout) (apiv1.MonoVertexRollout, error) {
+			mvr.Spec.MonoVertex.Spec.Raw = rawSpec
+			return mvr, nil
+		})
+
+		VerifyMonoVertexRolloutScaledDownForProgressive(monoVertexRolloutName, 0, monoVertexScaleTo, int64(monoVertexScaleMin), monoVertexScaleMinMaxJSONString, monoVertexScaleTo)
+		VerifyMonoVertexRolloutProgressiveStatus(monoVertexRolloutName, 0, 1, false, apiv1.AssessmentResultSuccess, strategy.Progressive.ForcePromote)
+
+		// Won't check running pods in this case since Numaflow does not create new pods if the previous batch is not ready
```

Comment:
```suggestion
                // Verify monovertex is scaled back up
		// Won't check running pods in this case since Numaflow does not create new pods if the previous batch is not ready
```